### PR TITLE
C++: Fix warning from compile-query.

### PR DESCRIPTION
**Pull request by criemen**
Fixes  
```Warning: WARNING: Selected entity does not provide a location. (/home/runner/work/semmle-code/semmle-code/ql/cpp/ql/src/Diagnostics/FailedExtractorInvocations.ql:22,8-9)```

### CODE REVIEW

File: cpp/ql/src/Diagnostics/FailedExtractorInvocations.ql [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /**
  * @name Failed extractor invocations
  * @description Gives the command line of compilations for which extraction did not run to completion.
- * @kind diagnostic
+ * @kind table
```
Comment (geoffw0) [inline cpp/ql/src/Diagnostics/FailedExtractorInvocations.ql:4]: Do you know what `@kind diagnostic` is supposed to be?  Some of them seem to have a 'severity' value in the third column of the `select`, while others don't have a third column.

**Comment by jbj** [comment]
@criemen is now on holiday and can't answer the question.

@adityasharad, can you shed some light on what might be going on in this PR and whether the new `@kind` looks right for this query?

File: cpp/ql/src/Diagnostics/FailedExtractorInvocations.ql (hunk 1 above)
Comment (adityasharad) [reply cpp/ql/src/Diagnostics/FailedExtractorInvocations.ql:4]: For diagnostic queries, all of the following result patterns are accepted:
- `message, <optional list of placeholder pairs>`
- `message, <optional list of placeholder pairs>, severity`
- `entity, message, <optional list of placeholder pairs>` (like a regular alert query)
- `entity, message, <optional list of placeholder pairs>, severity`

Looking at the warning you're seeing, if an `entity` is provided we do need it to have a location, and in this case `Compilation` is not `Locatable`. If you want to keep this query as `@kind diagnostic`, you could either use an entity with a location (if a suitable one exists), or use one of the first two formats in the list above.

**Comment by criemen** [comment]
I updated the PR by switching to the message-only template, and rebased to avoid a submodule conflict for the internal PR.

**Review by ngoclong19121990 — APPROVED** [review]

**Review by MathiasVP — APPROVED** [review]
LGTM!

**Pull request merged**